src/librustc/hir/lowering.rs

```diff
@@ -5298,13 +5298,23 @@ impl<'a> LoweringContext<'a> {
 
     fn elided_path_lifetime(&mut self, span: Span) -> hir::Lifetime {
         match self.anonymous_lifetime_mode {
-            // N.B., We intentionally ignore the create-parameter mode here
-            // and instead "pass through" to resolve-lifetimes, which will then
-            // report an error. This is because we don't want to support
-            // impl elision for deprecated forms like
+            // In create-parameter mode we error here because we don't want to support
+            // deprecated impl elision in new features like impl elision and `async fn`,
+            // both of which work using the `CreateParameter` mode:
             //
             //     impl Foo for std::cell::Ref<u32> // note lack of '_
-            AnonymousLifetimeMode::CreateParameter |
+            //     async fn foo(_: std::cell::Ref<u32>) { ... }
+            AnonymousLifetimeMode::CreateParameter => {
+                struct_span_err!(
+                    self.sess,
+                    span,
+                    E0726,
+                    "implicit elided lifetime not allowed here"
+                ).help("consider using an explicit elided lifetime: `'_`").emit();
+                self.sess.abort_if_errors();
```

Comment:
This seems pretty bad tho to fatally abort compilation; ideally we should let it continue... if that cannot be feasibly fixed now, let's please add a `FIXME` here with an associated `A-diagnostics` issue.

Reply:
Why is this bad? There's no reason this is needed-- we could certainly continue compilation, but elsewhere in lowering we abort on errors, so I did the same here.

Reply:
My understanding is that we generally want compilation to continue to let the user see other errors beyond the lowering stage; same with parsing, i.e. that's why we do recovery... but @estebank  is the expert here.

Reply:
We're removing as many of the `abort_if_errors` as possible in order to avoid [the "waves of errors" experience](https://mobile.twitter.com/isislovecruft/status/876542921323028480). The more gating steps in the compiler the more likely you are to have real errors hidden from the users. A more reasonable thing to do is either marking entire blocks with placeholders (`tcx.types.error` is a good example) and let the compiler continue and ignore those placeholders. It pollutes later stages with some logic to ignore them but it dramatically improves the experience. In this case in particular you're already recovering with `self.new_named_lifetime(id, span, hir::LifetimeName::Error)`, so `abort_if_errors` shouldn't be needed. Did you see any knock-down effects? It'd be much better to fix those places than having this here, but of course that could be left for a later clean up PR, which is why leaving a `FIXME` is enough in this case.

Reply:
Huh, I guess I'm one of the only remaining ones who *wants* the "waves of errors" experience. I've had many fewer problems with "waves of errors" than I have with getting hundreds of errors due to a failed import or a missing WF bound. I'll go ahead and remove this, though, since it seems like there's a plan and policy in place. Thanks for bringing this up, @Centril , and for clarifying the plan, @estebank ! :)

Reply:
@cramertj we certainly want to silence useless knock-down errors, but to do so we first need to see/experience them :)

I think long term it'll be a much better experience.

Reply:
Makes sense! I'll do my best to follow bugs when I see these so we can track them down in the future.